packages/notebook-extension/src/index.ts

```diff
@@ -1623,6 +1639,18 @@ function populateMenus(app: JupyterLab, mainMenu: IMainMenu, tracker: INotebookT
     }
   } as IFileMenu.ICloseAndCleaner<NotebookPanel>);
 
+  // Add a save with view command to the file menu.
+  mainMenu.fileMenu.persistAndSavers.add({
+    tracker,
+    action: 'with View State',
+    name: 'Notebook',
+    persistAndSave: (current: NotebookPanel) => {
+      NotebookActions.persistOutputsCollapsed(current.notebook);
+      app.commands.execute('docmanager:save');
+      return Promise.resolve();
```

Comment:
Is there a reason you are returning a resolved promise here, rather than `return app.comands.execute('docmanager:save');`?

Reply:
Nope. I have changed it.